library/core/src/fmt/mod.rs

```diff
@@ -400,10 +489,17 @@ impl<'a> Arguments<'a> {
     #[must_use]
     #[inline]
```

Comment:
Consider using `#[inline(always)]`? Currently. with `opt-level=z`, `as_str` is not always inlined: <https://godbolt.org/z/rzsG3fYEK>.